### PR TITLE
Allow to use mysql format without ssl

### PR DESCRIPTION
**Pull request by proller**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Build/Testing/Packaging Improvement

Short description (up to few sentences):
Allow to use mysql format without ssl

### CODE REVIEW

File: cmake/find/poco.cmake [hunk 1]
```diff
@@ -8,6 +8,14 @@ if (NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/poco/CMakeLists.txt")
    set (MISSING_INTERNAL_POCO_LIBRARY 1)
 endif ()
 
+if (NOT ENABLE_LIBRARIES)
+    set (ENABLE_POCO_NETSSL ${ENABLE_LIBRARIES} CACHE BOOL "")
```
Comment (abyss7) [inline cmake/find/poco.cmake]: This doesn't work for CMake ≤ 3.13 - it doesn't respect variables before options with a same name.

File: dbms/programs/server/MySQLHandler.cpp [hunk 2]
```diff
@@ -28,24 +30,33 @@ namespace DB
...
 namespace ErrorCodes
 {
     extern const int MYSQL_CLIENT_INSUFFICIENT_CAPABILITIES;
     extern const int OPENSSL_ERROR;
+    extern const int SUPPORT_IS_DISABLED;
 }
 
-MySQLHandler::MySQLHandler(IServer & server_, const Poco::Net::StreamSocket & socket_, RSA & public_key_, RSA & private_key_, bool ssl_enabled, size_t connection_id_)
+MySQLHandler::MySQLHandler(IServer & server_, const Poco::Net::StreamSocket & socket_,
+#if USE_SSL
```
Comment (abyss7) [inline dbms/programs/server/MySQLHandler.cpp]: It's a bad design to have a different interface depending on macro - better to have derived class.
Same for every occurrence below.

File: dbms/programs/server/MySQLHandler.h [hunk 3]
```diff
@@ -16,7 +17,11 @@ namespace DB
 class MySQLHandler : public Poco::Net::TCPServerConnection
 {
 public:
-    MySQLHandler(IServer & server_, const Poco::Net::StreamSocket & socket_, RSA & public_key_, RSA & private_key_, bool ssl_enabled, size_t connection_id_);
+    MySQLHandler(IServer & server_, const Poco::Net::StreamSocket & socket_,
+#if USE_SSL
```
Comment (abyss7) [inline dbms/programs/server/MySQLHandler.h]: +

**Review by abyss7 — CHANGES_REQUESTED** [review]

File: cmake/find/poco.cmake (hunk 1 above)
Comment (proller) [reply cmake/find/poco.cmake]: Even with CACHE BOOL "" ?

File: dbms/programs/server/CMakeLists.txt [hunk 4]
```diff
@@ -10,13 +10,11 @@ set(CLICKHOUSE_SERVER_SOURCES
     ${CMAKE_CURRENT_SOURCE_DIR}/TCPHandler.cpp
    )
 
-if (USE_SSL)
     set(CLICKHOUSE_SERVER_SOURCES
```
Comment (alexey-milovidov) [inline dbms/programs/server/CMakeLists.txt]: Wrong indent.

File: dbms/programs/server/MySQLHandler.cpp [hunk 5]
```diff
@@ -212,6 +224,9 @@ void MySQLHandler::finishHandshake(MySQLProtocol::HandshakeResponse & packet)
         packet_sender = std::make_shared<PacketSender>(*in, *out, connection_context.mysql.sequence_id);
         packet_sender->max_packet_size = connection_context.mysql.max_packet_size;
         packet_sender->receivePacket(packet); /// Reading HandshakeResponse from secure socket.
+#else
+        throw Exception("Compiled without ssl", ErrorCodes::SUPPORT_IS_DISABLED);
```
Comment (alexey-milovidov) [inline dbms/programs/server/MySQLHandler.cpp]: SSL must be in upper case.

File: dbms/programs/server/MySQLHandler.cpp [hunk 6]
```diff
@@ -232,7 +247,13 @@ void MySQLHandler::authenticate(const String & user_name, const String & auth_pl
     // For compatibility with JavaScript MySQL client, Native41 authentication plugin is used when possible (if password is specified using double SHA1). Otherwise SHA256 plugin is used.
     auto user = connection_context.getUser(user_name);
     if (user->authentication.getType() != DB::Authentication::DOUBLE_SHA1_PASSWORD)
+    {
+#if USE_SSL
         auth_plugin = std::make_unique<MySQLProtocol::Authentication::Sha256Password>(public_key, private_key, log);
+#else
+        throw Exception("Compiled without ssl", ErrorCodes::SUPPORT_IS_DISABLED);
```
Comment (alexey-milovidov) [inline dbms/programs/server/MySQLHandler.cpp]: SSL must be in upper case.

**Comment by alexey-milovidov** [comment]
Superseded by https://github.com/ClickHouse/ClickHouse/pull/7524

**Pull request closed without merging**